Lib/multiprocessing/pool.py

```diff
@@ -92,7 +92,14 @@ def __repr__(self):
 
 def worker(inqueue, outqueue, initializer=None, initargs=(), maxtasks=None,
            wrap_exception=False):
-    assert maxtasks is None or (type(maxtasks) == int and maxtasks > 0)
+    if maxtasks is not None:
```

Comment:
`worker()` is an internal API, I'm not sure it's worth maintaining this more complicated error reporting.